athena-opensearch/src/main/java/com/amazonaws/athena/connectors/opensearch/OpensearchRecordHandler.java

```diff
@@ -0,0 +1,111 @@
+import static com.amazonaws.athena.connectors.opensearch.OpensearchConstants.OPENSEARCH_NAME;
+import static com.amazonaws.athena.connectors.opensearch.OpensearchConstants.OPENSEARCH_QUOTE_CHARACTER;;
+
+
+/**
+ * Data handler, user must have necessary permissions to read from necessary tables.
+ */
+public class OpensearchRecordHandler
+        extends JdbcRecordHandler
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpensearchRecordHandler.class);
+
+    private static final String OPENSEARCH_QUOTE_CHARACTER = "`";
+
+    private final JdbcSplitQueryBuilder jdbcSplitQueryBuilder;
+
+    /**
+     * Instantiates handler to be used by Lambda function directly.
+     *
+     * Recommend using {@link OpensearchMuxCompositeHandler} instead.
+     */
+    public OpensearchRecordHandler(java.util.Map<String, String> configOptions)
+    {
+        this(JDBCUtil.getSingleDatabaseConfigFromEnv(OPENSEARCH_NAME, configOptions), configOptions);
+    }
+
+    public OpensearchRecordHandler(DatabaseConnectionConfig databaseConnectionConfig, java.util.Map<String, String> configOptions)
+    {
+        this(databaseConnectionConfig, new GenericJdbcConnectionFactory(databaseConnectionConfig, OpensearchMetadataHandler.JDBC_PROPERTIES, new DatabaseConnectionInfo(OPENSEARCH_DRIVER_CLASS, OPENSEARCH_DEFAULT_PORT)), configOptions);
+    }
+
+    public OpensearchRecordHandler(DatabaseConnectionConfig databaseConnectionConfig, JdbcConnectionFactory jdbcConnectionFactory, java.util.Map<String, String> configOptions)
+    {
+        this(databaseConnectionConfig, AmazonS3ClientBuilder.defaultClient(), AWSSecretsManagerClientBuilder.defaultClient(), AmazonAthenaClientBuilder.defaultClient(),
+                jdbcConnectionFactory, new OpensearchQueryStringBuilder(OPENSEARCH_QUOTE_CHARACTER, new OpensearchFederationExpressionParser(OPENSEARCH_QUOTE_CHARACTER)), configOptions);
+    }
+
+    @VisibleForTesting
+    OpensearchRecordHandler(DatabaseConnectionConfig databaseConnectionConfig, final AmazonS3 amazonS3, final AWSSecretsManager secretsManager,
+            final AmazonAthena athena, JdbcConnectionFactory jdbcConnectionFactory, JdbcSplitQueryBuilder jdbcSplitQueryBuilder, java.util.Map<String, String> configOptions)
+    {
+        super(amazonS3, secretsManager, athena, databaseConnectionConfig, jdbcConnectionFactory, configOptions);
+        this.jdbcSplitQueryBuilder = Validate.notNull(jdbcSplitQueryBuilder, "query builder must not be null");
+    }
+
+    @Override
+    public PreparedStatement buildSplitSql(Connection jdbcConnection, String catalogName, TableName tableName, Schema schema, Constraints constraints, Split split)
+            throws SQLException
+    {
+        PreparedStatement preparedStatement = jdbcSplitQueryBuilder.buildSql(jdbcConnection, null, null, tableName.getTableName(), schema, constraints, split);
+
+        // Disable fetching all rows.
+        preparedStatement.setFetchSize(10000);
```

Comment:
let's put this in a constant.

what happens if the input query has over 10k results?

Reply:
Setting fetch size `Gives the JDBC driver a hint as to the number of rows that should be fetched from the database when more rows are needed for ResultSet objects generated by this Statement`.
So what this really controls is the number of network calls, as well as enforces pagination so that SQL Plugin uses v2 instead of legacy engine. Otherwise legacy version only returns 200 rows per table.